mico-core/src/main/java/io/github/ust/mico/core/broker/MicoApplicationBroker.java

```diff
@@ -145,19 +154,51 @@ public MicoApplication copyAndUpgradeMicoApplicationByShortNameAndVersion(String
         // we need to set the id of all those nodes to null. That way, Neo4j will create new entities
         // instead of updating the existing ones.
         micoApplication.setId(null);
-        micoApplication.getServiceDeploymentInfos().forEach(sdi -> sdi.setId(null));
-        micoApplication.getServiceDeploymentInfos().forEach(sdi -> sdi.getLabels().forEach(label -> label.setId(null)));
-        micoApplication.getServiceDeploymentInfos().forEach(sdi -> {
-            sdi.getEnvironmentVariables().forEach(envVar -> envVar.setId(null));
-            sdi.getInterfaceConnections().forEach(ic -> ic.setId(null));
-        });
-        // The actual Kubernetes deployment information must not be copied, because the new application
-        // is considered to be not deployed yet.
-        micoApplication.getServiceDeploymentInfos().forEach(sdi -> sdi.setKubernetesDeploymentInfo(null));
+        prepareServiceDeploymentInfosForCopying(micoApplication.getServiceDeploymentInfos());
+        prepareServiceDeploymentInfosForCopying(micoApplication.getKafkaFaasConnectorDeploymentInfos().stream()
+            .map(KFConnectorDeploymentInfo::getServiceDeploymentInfo).collect(Collectors.toList()));
 
         return createMicoApplication(micoApplication);
     }
 
+    /**
+     * Retrieves the list of {@code KFConnectorDeploymentInfos} that are part of the {@code MicoApplication}.
```

Comment:
Add something referencing the kafka faas connector to this javadoc to help explain the abbreveation (also valid for other kfc instances...)